src/options.js

```diff
@@ -70,6 +74,104 @@ function loadOptions() {
 $(loadOptions);
 
 /**
+ * Import a list of trackers supplied by the user
+ * NOTE: list must be in JSON format to be parseable
+ */
+function importTrackerList() {
+  var file = $('#inputFile')[0].files[0];
+
+  if (file) {
+    var reader = new FileReader();
+    reader.readAsText(file);
+    reader.onload = function(e) {
+      parseTrackerList(e.target.result);
+    };
+  } else {
+    confirm("Please select a file to import.");
+  }
+
+  document.getElementById("inputFile").value = '';
+}
+
+/**
+ * Parse the tracker lists uploaded by the user, adding to
+ * action_map and snitch_map anything that isn't currently present.
+ *
+ * @param trackerLists - data from JSON file that user provided
+ */
+function parseTrackerList(trackerLists) {
+  var lists;
+
+  try {
+    lists = JSON.parse(trackerLists);
+  } catch (e) {
+    confirm("File must be of type .json.");
+  }
+
+  for (var i in lists) {
+    var elem = lists[i];
+    if (elem.action_map) {
+      updateMap(elem.action_map, "action_map");
+    } else if (elem.snitch_map) {
+      updateMap(elem.snitch_map, "snitch_map");
```

Comment:
Import settings map as well. Refactor JSON output to one object instead of array. 

```
{
   "snitch_map": {},
   "action_map": {},
   ...
}
```

```
for key in input:
   updateMap(value, key);
```